### PR TITLE
add bandit, fix vulnerabilities & bump pre-commit-config

### LINKED ISSUE #61: Set up bandit
Add https://bandit.readthedocs.io/en/latest/ to pre-commit

### PR DESCRIPTION
**Pull request by vvssttkk**
close #61
continue #424

### CODE REVIEW

**Comment by aguschin** [comment]
@vvssttkk, fixed few failed tests I was talking about. Please investigate others. E.g. one of the frequent failures I reproduced in 3.7:

```
tmp %f $ python                                                                                                                                   181ms
Python 3.7.10 (default, Jun 25 2021, 16:52:37)
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import hashlib
>>> hashlib.md5()
<md5 HASH object @ 0x109a1bdb0>
>>> hashlib.md5(userforsecurity=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: openssl_md5() takes no keyword arguments
>>>
```

File: mlem/utils/templates.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ def generate(self, **additional):
         j2 = Environment(
             loader=FileSystemLoader(self.templates_dir + [self.TEMPLATE_DIR]),
             undefined=StrictUndefined,
+            autoescape=True,
```
Comment (aguschin) [inline mlem/utils/templates.py:25]: One more example, `test_dockerfile_generator_super_custom`. It has to do something with your changes, cause I just run CI in a nearby PR #443 and it worked for me. I assume it is this change. Maybe we need to fix the test (if everything works if you `docker build` that custom dockerfile - have to check it), maybe we need to disable the check here (if this is ok to not escape things in dockerfiles). Relying on you to find this out :)

**Comment by aguschin** [comment]
Nice! Looks like a black formatting fails, but everything else works!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -54,7 +54,11 @@ repos:
...
-  #   hooks:
-  #     - id: bandit
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.4
+    hooks:
+      - id: bandit
+        exclude: tests/
+        args:
+          - -iii  # high level
+          - -lll  # high confidense
```
Comment (madhur-tandon) [inline .pre-commit-config.yaml:64]: `confidense` --> `confidence`

**Comment by aguschin** [comment]
Great! Thank you @vvssttkk! Merging this! 🚀

**Comment by mertbozkir** [comment]
Hello @vvssttkk, how can I get in touch with you? Can you reach me from any platform? We have something for you! 🤓

**Pull request merged**